torch/testing/__init__.py

```diff
@@ -87,9 +87,15 @@ def make_non_contiguous(tensor):
 
 
 def get_all_dtypes():
+    return [torch.uint8, torch.bool, torch.int8, torch.int16, torch.int32, torch.int64,
+            torch.float16, torch.float32, torch.float64]
+
+def get_all_math_dtypes():
```

Comment:
this needs to take a device -- torch.float16 is a math dtype on cuda but not cpu.